app/validators/routes_and_redirects_validator.rb

```diff
@@ -147,11 +147,37 @@ def validate(location, redirect)
 
   private
 
+    def acceptable_destination?(target)
+      target.starts_with?("/") || is_valid_govuk_campaign_url?(target)
+    end
+
+    def is_valid_govuk_campaign_url?(target)
+      uri = URI.parse(target)
+      host = uri.host
+      if (host =~ /\A.+\.campaign\.service\.gov\.uk\z/i) != nil &&
+          %w(http https).include?(uri.scheme)
+        label = host&.split(".")&.first
```

Comment:
Well in the tests you will notice a set of malformed urls.. In this instance say `http://.campaign.gov.uk/path` The intention here is to split by dot which will return `["", "campaign", "gov", "uk"]`, so the first would not in this case suffice..